keyboards/dz60/keymaps/rundeploy/keymap.c

```diff
@@ -0,0 +1,178 @@
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_HOME, KC_PGDOWN, KC_END),
+
+
+  /* MO(2): Macro Layer
+   * ,----------------------------------------------------------------.
+   * |SysO|  |   |   |   |   |   |   |   |   |   |   |   |     |FrwDel|
+   * |----------------------------------------------------------------|
+   * |     |TOG|MOD|HUI|HUD|SAI|SAD|VAI|VAD|   |   |   |   |   DFU    |
+   * |----------------------------------------------------------------|
+   * |       |PLN|BRT|RNB|SWR|SNK|KNG|XMS|GRD|   |   |   |            |
+   * |----------------------------------------------------------------|
+   * |        |   |   |BL-| BL|BL+|BLS|   |   |   |   |   |     |     |
+   * |----------------------------------------------------------------|
+   * |    |     |    |        |    |          |   |   |   |     |     |
+   * `----------------------------------------------------------------'
+   */
+  [_MACRO] = LAYOUT_all(
+    KC_POWER, M(1), M(2), M(3), M(4), M(5), M(6), M(7), M(8), M(9), M(10), M(11), M(12), KC_TRNS, KC_DEL,
+    KC_TRNS, RGB_TOG, RGB_MOD, RGB_HUI, RGB_HUD, RGB_SAI, RGB_SAD, RGB_VAI, RGB_VAD, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, RESET,
+    KC_TRNS, RGB_MODE_PLAIN, RGB_MODE_BREATHE, RGB_MODE_RAINBOW, RGB_MODE_SWIRL, RGB_MODE_SNAKE, RGB_MODE_KNIGHT, RGB_MODE_XMAS, RGB_MODE_GRADIENT, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_NO,
+    KC_TRNS, KC_NO, KC_TRNS, KC_TRNS, BL_DEC, BL_TOGG, BL_INC, BL_STEP, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS),
+};
+
+/*----------------------------------MACRO------------------------------------*/
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  switch(keycode) {
+    case EM1:
+      if (record->event.pressed) {
+        SEND_STRING("mail1@gmail.com"); //change to your email
+      }
+    return false;
+    break;
+
+    case EM2:
+      if (record->event.pressed) {
+        SEND_STRING("mail2@gmail.com"); //change to your email
+      }
+    return false;
+    break;
+
+    case PS1:
+      if (record->event.pressed) {
+        SEND_STRING("pass1"); //change to your pass
+      }
+    return false;
+    break;
+
+    case PS2:
+      if (record->event.pressed) {
+        SEND_STRING("pass2"); //change to your pass
+      }
+    return false;
+    break;
+
+    case NNM:
+      if (record->event.pressed) {
+        SEND_STRING("pass2"); //change to your nickname
+      }
+    return false;
+    break;
+  }
+  return true;
+};
+
+/*----------------------------------------------------------------------------------*/
+enum function_id{
+    SHIFT_ESC,
+};
+
+const uint16_t PROGMEM fn_actions[] = {
+  [0]  = ACTION_FUNCTION(SHIFT_ESC),
+};
+
+void action_function(keyrecord_t *record, uint8_t id, uint8_t opt) {
```

Comment:
This looks like it's performing the same function as Grave Escape. 

If so, it may be better to use that, as functions and "action_macro" have been depreciated in favor of `process_record_user`. 

https://docs.qmk.fm/#/feature_grave_esc

https://docs.qmk.fm/#/feature_macros